### ISSUE TITLE
DotNetCore Preparation - Abstract Examine

### ISSUE THREAD

**Issue by anthonydotnet** [umbraco/Umbraco-CMS#5628]
As discussed at Codegarden 2019, to transition to dotnet core, Umbraco requires significant refactoring.

This task is to abstract and decouple Examine.

@Shazwazza

**Comment by Shazwazza** [comment]
Hi @anthonydotnet , I think this task will prob need to exist under Examine itself and/or have another task here for updating the reference ... though to update the reference will require that Examine 2.0 (or whatever version it will be) will have something available on MyGet (or similar).

Just notes on what this task actually means:

As part of the [project split phase of project UniCore](https://github.com/umbraco/rfcs/pull/6) we want the `Umbraco.Examine` project to only have references to Examine's abstractions (so it's not also pulling in Lucene.Net). This means splitting up the Examine project into 2 packages: Examine.Core and Examine.Lucene (probably) where Examine.Core is all abstractions and doesn't have dependencies on anything. Then `Umbraco.Examine` only references `Examine.Core` 

I'm not sure @nul800sebastiaan will want to keep these tasks open if they aren't yet scheduled. I think we have a tag of some sort for that but will let you guys chat about that.

**Comment by Shazwazza** [comment]
related: #5631, #5630, #5629, #5627

**Comment by nul800sebastiaan** [comment]
Hey @anthonydotnet - cool that you're being very pro-active but unfortunately we're not ready to move on these items yet. 

As explained in the roadmap overview that Jacob showed during the CG19 keynote, this is a future plan and we're not working on this right now, we have other priorities first. I'm sure you're eager to work on this but we have yet to flesh out the full plans. 

I'm happy for a discussion to continue on Our Umbraco if this is something the community needs to talk about at this point but it's a little premature at this point. Once there's a plan, we can re-open this and the related issues with the details of the work required.

Thanks!

**Comment by anthonydotnet** [comment]
> Hey @anthonydotnet - cool that you're being very pro-active but unfortunately we're not ready to move on these items yet.
> 
> As explained in the roadmap overview that Jacob showed during the CG19 keynote, this is a future plan and we're not working on this right now, we have other priorities first. I'm sure you're eager to work on this but we have yet to flesh out the full plans.
> 
> I'm happy for a discussion to continue on Our Umbraco if this is something the community needs to talk about at this point but it's a little premature at this point. Once there's a plan, we can re-open this and the related issues with the details of the work required.
> 
> Thanks!


These issues were discussed with @Shazwazza at the open circle at CG. He said that these tasks could be worked on by the community, and that I should create issues on the issue tracker for them. 

Is this no longer the case?

**Comment by Shazwazza** [comment]
@anthonydotnet from what i recall you said you were going to host a global hackathon and wanted tasks to work on for that, so we came up with some items that could be done 'now' by the community outside of the HQ schedule. If you are going to do that hackathon or are going to actively work on these, then sure please have tasks here to commit/reference against as you work through a PR otherwise we can re-open these when they are scheduled.

**Comment by anthonydotnet** [comment]
Yep. The plan is to arrange a hackathon to try to tackle these.

I would suggest keeping these open too, as there may be people out there who would like to help in their spare time.

**Comment by nul800sebastiaan** [comment]
The RFC is not finalized yet, so we haven't made these decisions yet. It's great that you want to work on something bigger but it needs to be done with the guidance of HQ, we wouldn't want to waste anybody's time of them working on these items and then coming to the conclusion it's not the way to go.

For anybody to be able to work on this feature, we would need to provide details so that people know what to do, where to decouple it to, in what way, etc. 

Currently, we're awaiting the result of the RFC before we can provide guidance on these items. Once we have that, we can update the related issues and re-open them.
